cmake/CompileBoost.cmake

```
@@ -1,12 +1,12 @@
-find_package(Boost 1.67)
+find_package(Boost 1.72)
 
 if(Boost_FOUND)
   add_library(boost_target INTERFACE)
   target_link_libraries(boost_target INTERFACE Boost::boost)
 else()
   include(ExternalProject)
   ExternalProject_add(boostProject
-    URL "https://dl.bintray.com/boostorg/release/1.67.0/source/boost_1_67_0.tar.bz2"
+    URL "https://dl.bintray.com/boostorg/release/1.72.0/source/boost_1_72_0.tar.bz2"
     URL_HASH SHA256=2684c972994ee57fc5632e03bf044746f6eb45d4920c343937a465fd67a5adba
```

Used correct hash for Boost 1.72
```suggestion
    URL_HASH SHA256=59c9b274bc451cf91a9ba1dd2c7fdcaf5d60b1b3aa83f2c9fa143417cc660722
```